Tests/BotTests/GitHub/GitHubEventsTests.swift

```diff
@@ -2,83 +2,164 @@ import XCTest
+                let req = request(
+                    withRawEvent: "anything_really",
+                    signature: "9f64868b5cd91faa4c63589acff7286b16d289ae",
+                    payload: "{}"
+                )
+
+                let result = service.handleEvent(from: req).first()
+
+                expect(result?.error) == .unknown
+            },
+            assert: { events in
+                expect(events) == []
+            }
+        )
+    }
 
-        let result = service.handleEvent(from: request).first()
+    func test_handling_untrustworthy_payload() {
+        perform(
+            when: { service, scheduler in
 
-        scheduler.advance()
+                let req = request(
+                    withRawEvent: "anything_really",
+                    signature: "00faadd5dc1395b616a546ac6f67796409b4f839",
+                    payload: "{}"
+                )
 
-        expect(result?.error).to(beNil())
-        expect(observedEvents) == [Event.pullRequest(pullRequestEvent)]
+                let result = service.handleEvent(from: req).first()
+
+                expect(result?.error) == .untrustworthy
+            },
+            assert: { events in
+                expect(events) == []
+            }
+        )
     }
+}
 
-    func test_handling_ping_event() {
-        let service = GitHubService()
-        let scheduler = TestScheduler()
+extension GitHubEventsTests {
+    
+    static let signatureToken = "Wall-E"
 
-        let request = StubbedRequest(
-            headers: [GitHubService.HTTPHeader.event.rawValue: GitHubService.APIEvent.ping.rawValue],
-            body: nil
+    private var stubbedPullRequestEvent: PullRequestEvent {
+        return PullRequestEvent(
+            action: .closed,
+            pullRequestMetadata: PullRequestMetadata(
+                reference: PullRequest(
+                    number: 1,
+                    title: "Update the README with new information",
+                    author: PullRequest.Author(login: "Codertocat"),
+                    source: PullRequest.Branch(ref: "changes", sha: "34c5c7793cb3b279e22454cb6750c80560547b3a"),
+                    target: PullRequest.Branch(ref: "master", sha: "a10867b14bb761a232cd80139fbd4c0d33264240"),
+                    labels: [.init(name: "bug")]
+                ),
+                isMerged: false,
+                mergeState: .clean
+            )
         )
+    }
+
+    private func perform(
+        stub: (TestScheduler) -> GitHubService = stubService,
+        when: (GitHubService, TestScheduler) -> Void,
+        assert: ([Event]) -> Void
+        ) {
+
+        let scheduler = TestScheduler()
+
+        let service = stub(scheduler)
 
         var observedEvents: [Event] = []
 
         service.events
             .observe(on: scheduler)
             .observeValues { event in observedEvents.append(event) }
 
-        let result = service.handleEvent(from: request).first()
+        when(service, scheduler)
```

Comment:
Maybe make `let result = when(service, scheduler)` then pass the result to `assert` too, so that the assertions on `result.error` can be moved to the assert closure. Would fit better with the Arrange/Act/Assert pattern